doc/source/whatsnew/v1.3.0.rst

```diff
@@ -217,6 +217,8 @@ Numeric
 - Bug in :meth:`DataFrame.select_dtypes` with ``include=np.number`` now retains numeric ``ExtensionDtype`` columns (:issue:`35340`)
 - Bug in :meth:`DataFrame.mode` and :meth:`Series.mode` not keeping consistent integer :class:`Index` for empty input (:issue:`33321`)
 - Bug in :meth:`DataFrame.rank` with ``np.inf`` and mixture of ``np.nan`` and ``np.inf`` (:issue:`32593`)
+- Bug in :meth:`DataFrame.rank` with ``axis=0`` and columns holding incomparable types raising ``IndexError``
```

Comment:
if there is not issue (its fine don't create one), but add this PR number as the issue number.

Reply:
Thanks for committing that, yep no issue